scanners/ssh-audit/parser/parser.test.js

```diff
@@ -0,0 +1,587 @@
+              "hash": "c8:65:6b:d1:59:03:56:21:d9:0f:84:83:ce:ac:40:86",
+              "hash_alg": "MD5",
+              "hostkey": "ssh-ed25519",
+            },
+            {
+              "hash": "MbRX/CgQyN6/p8/ZjORurfaJqDhu4VEIWfXo0BnxaCE",
+              "hash_alg": "SHA256",
+              "hostkey": "ssh-rsa",
+            },
+            {
+              "hash": "a5:6f:62:26:81:03:b7:5e:06:48:10:04:79:4b:ac:32",
+              "hash_alg": "MD5",
+              "hostkey": "ssh-rsa",
+            },
+          ],
+          "host": "dummy-ssh.demo-targets.svc",
+          "key_algorithms": [
+            {
+              "algorithm": "ssh-rsa",
+              "keysize": 2048,
+            },
+            {
+              "algorithm": "rsa-sha2-512",
+              "keysize": 2048,
+            },
+            {
+              "algorithm": "rsa-sha2-256",
+              "keysize": 2048,
+            },
+            {
+              "algorithm": "ecdsa-sha2-nistp256",
+            },
+            {
+              "algorithm": "ssh-ed25519",
+            },
+          ],
+          "key_exchange_algorithms": [
+            {
+              "algorithm": "curve25519-sha256@libssh.org",
+            },
+            {
+              "algorithm": "ecdh-sha2-nistp256",
+            },
+            {
+              "algorithm": "ecdh-sha2-nistp384",
+            },
+            {
+              "algorithm": "ecdh-sha2-nistp521",
+            },
+            {
+              "algorithm": "diffie-hellman-group-exchange-sha256",
+              "keysize": 2048,
+            },
+            {
+              "algorithm": "diffie-hellman-group14-sha1",
+            },
+          ],
+          "mac_algorithms": [
+            "umac-64-etm@openssh.com",
+            "umac-128-etm@openssh.com",
+            "hmac-sha2-256-etm@openssh.com",
+            "hmac-sha2-512-etm@openssh.com",
+            "hmac-sha1-etm@openssh.com",
+            "umac-64@openssh.com",
+            "umac-128@openssh.com",
+            "hmac-sha2-256",
+            "hmac-sha2-512",
+            "hmac-sha1",
+          ],
+          "server_banner": "SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.8",
+          "ssh_lib_cpe": "OpenSSH_7.2p2",
+          "ssh_version": 2,
+        },
+        "category": "SSH Service",
+        "description": "Information about Used SSH Algorithms",
+        "location": "dummy-ssh.demo-targets.svc",
+        "name": "SSH Service",
+        "osi_layer": "APPLICATION",
+        "severity": "INFORMATIONAL",
+      },
+      {
+        "algorithms": [
+          "diffie-hellman-group14-sha1",
+          "ecdh-sha2-nistp256",
+          "ecdh-sha2-nistp384",
+          "ecdh-sha2-nistp521",
+        ],
+        "category": "SSH Policy Violation",
+        "description": "Discouraged SSH key exchange algorithms in use",
+        "hint": "Remove these KEX algorithms",
+        "name": "Insecure SSH KEX Algorithms",
+        "severity": "HIGH",
+      },
```

Comment:
At the moment these findings are not really usable (at least for users of the elasticsearch provider) as they don't contain a location. 
So the users could not tell which ssh server is actually using these discouraged kex algs.
All findings should have a location set (as done for the "SSH Service" finding)